modules/prebidServerBidAdapter/index.js

```diff
@@ -539,10 +539,13 @@ const OPEN_RTB_PROTOCOL = {
                 }
                 if (Array.isArray(params.aspect_ratios)) {
                   // pass aspect_ratios as ext data I guess?
-                  asset.ext = {
-                    aspectratios: params.aspect_ratios.map(
-                      ratio => `${ratio.ratio_width}:${ratio.ratio_height}`
-                    )
+                  const aspectRatios = params.aspect_ratios
+                    .filter((ar) => ar.ratio_width && ar.ratio_height)
+                    .map(ratio => `${ratio.ratio_width}:${ratio.ratio_height}`);
+                  if (aspectRatios.length === params.aspect_ratios.length) {
```

Comment:
hey @dgirardi, can you help me understand why is this condition comparing length of the original aspect_ratios array with filtered/mapped aspectRatios? Shouldn't we just check if `aspectRatios.length > 0`?

Also L534 and L535 are reading wmin and hmin from the first aspect_ratios element, is this valid given that aspect_ratios is an array, and that there might be some other elements of that array with min_width & min_height?

Reply:
well, sizes property of the native asset is not really meant to be an array, at least according to the ORTB, even though it's defined as such in the pbjs docs. Since prebid server adapter is translating prebid config to ORTB request, where image asset can have only one wmin and hmin, I can only assume that sizes & aspect_ratios are defined as arrays to leave a possibility for other adapters that are not tied to the ORTB spec. 

When testing with only sizes defined for img asset (without aspect_ratios), this is what I found:

sizes: [989, 742] - works as intended
sizes: [[989, 742]] - does not work (error creating native request: invalid img sizes (must provide sizes or min_height & min_width if using aspect_ratios))
sizes: [[989, 742], [981, 741]] - works but not as intended (outgoing request contains "w":[3000,2250],"h":[3001,2251])

But this is another issue with native in pbs adapter, and there are many more in the way native is handled by pbs adapter, I'd say. So to address the actual issue this PR tries to solve, in the outgoing request pbs adapter makes, the only array in the img asset object is the `ext.aspectratios`, so there's nothing in this asset that can correspond 1:1 to this array, that's why I suggest we replace the restriction `aspectRatios.length === params.aspect_ratios.length` with `... > 0`.

Reply:
My instinct is that's not right, but I can't verify without seeing how `ext.aspectratios` is used out in the wild right now, and of course it was not right to begin with so I can't object too much.

The reason is so far if `ext.aspectratios` was present, the receiver could look at `ext.aspectratios[0]` to see what the aspect ratio is (and realistically, if anyone is looking at this field right now, that's what they have to be doing - the rest of the array wouldn't make sense as far as I can tell). That would no longer necessarily be true if we use `> 0` here; I'd expect it to break consumers if there are any - but I'll trust your judgement on the impact of this change.

Stepping back, if in ORTB2 there's really no place for more than one dimension, passing an array here doesn't make sense, does it? what do you think about adding `ext.ratio_width` and `ext.ratio_height` instead (or in addition)?

Reply:
Not sure what receiver you are referencing here, receiver of the request object or the response object? :) If we're talking about request, why would they look exactly at `ext.aspectratios[0]`, it's an array so it can offer more than one aspect ratio to the buy side? It anyway doesn't make sense to set `ext.aspectratios` if you have specified `w` & `h`, that's a hard requirement, to me it only makes sense in combination with `hmin` and `wmin`, but again, `ext.aspectratios` is not something that exists in ORTB standard, so I can only guess how someone uses it (or if it's being used at all).

If we're talking about the response, we have an image response object ([page 22](https://iabtechlab.com/wp-content/uploads/2016/07/OpenRTB-Native-Ads-Specification-Final-1.2.pdf)), with `w` & `h` and if aspect ratio is needed, it can be figured out based on w&h alone w/gcd(w, h) : h/gcd(w, h).

Regarding your suggestion for adding `ext.ratio_width` and `ext.ratio_height`, it is a bit better fit for the current ORTB structure (it can combined with wmin and hmin), but I'm not sure if anybody on the buy side would be consuming this (obviously nobody now, since nobody would know about it).

Anyway, to go back to the original remark I had on this PR, I would like to consult a second reviewer if he also thinks that this condition should be as is `aspectRatios.length === params.aspect_ratios.length`, or `... > 0`, and then we can merge this one.

Reply:
I'm not too sure how this is used in the wild, outside of the issue that was created around it either. I would lean a bit more toward `aspectRatios.length > 0` though because I think that is probably less likely to break various possible implementations. I wouldn't say I am an expert on this either though.